.github/workflows/commit.yml

```diff
@@ -264,3 +222,38 @@ jobs:
           COSIGN_PASSWORD: ${{ secrets.COSIGN_PASSWORD }}
           TOOL: ${{ matrix.tool }}
           ENVIRONMENT: ${{ matrix.environment }}
+  finalize-the-release:
+    name: Finalize the release
+    needs: [bump-version, distribute]
+    if: "${{ github.event_name == 'push' && !startsWith(github.event.head_commit.message, 'Bump version: 2') }}"
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: "${{ needs.bump-version.outputs.git_tag }}"
```

Comment:
This is extremely important to ensure that the release is of the post-auto-updated tag, not the initial commit that went to `main`.